src/server/index.tsx

```diff
@@ -46,6 +49,54 @@ server.get("/robots.txt", async (_req, res) => {
   res.send(robotstxt);
 });
 
+server.get("/css/themes-list", async (req, res) => {
+  const builtinThemes = [
+    "litera",
+    "materia",
+    "minty",
+    "solar",
+    "united",
+    "cyborg",
+    "darkly",
+    "journal",
+    "sketchy",
+    "vaporwave",
+    "vaporwave-dark",
+    "i386",
+    "litely",
+    "nord",
+  ];
+  fs.readdir(extraThemesFolder, function (_, data) {
+    if (data != null) {
+      data = data
+        .filter(d => d.endsWith(".min.css"))
+        .map(d => d.replace(".min.css", ""));
+      data = builtinThemes.concat(data);
+      // use set to remove duplicate values
+      data = Array.from(new Set(data));
+      res.send(data);
+    } else {
+      res.send(builtinThemes);
+    }
+  });
+});
+
+server.get("/css/themes/:name", async (req, res) => {
+  res.contentType("text/css");
+  const theme = req.params.name;
+  if (!theme.endsWith(".min.css")) {
```

Comment:
It might be confusing that themes wont work unless the name ends with .min.css. Maybe better to remove the .min part from builtin themes.

Reply:
Hrm.... I spose it should be able to read any css file, these are custom themes after all, and nothings going in this repo anyway. So not too necessary to do this check.

Reply:
.min.css is hardcoded here: https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/components/app/theme.tsx#L20